daemon/cluster/executor/container/adapter.go

```diff
@@ -39,7 +39,7 @@ func newContainerAdapter(b executorpkg.Backend, task *api.Task) (*containerAdapt
 
 func (c *containerAdapter) pullImage(ctx context.Context) error {
 	// if the image needs to be pulled, the auth config will be retrieved and updated
-	encodedAuthConfig := c.container.spec().RegistryAuth
+	encodedAuthConfig := c.container.spec().PullOptions.RegistryAuth
```

Comment:
add nil check like in https://github.com/docker/swarmkit/commit/2890e6dde066f46787d33b8fa8684c2f4555b76f#diff-91c312b84ef212e7a7675e103cb76d41R45